### DISCUSSION TITLE
Now With Translations!

### DISCUSSION THREAD

**Discussion by dannagle** [dannagle/PacketSender#313] [General]
# Now With Translations!

## - <a href="#french">Français</a>

<a id="english"></a>
<hr>
<img width="200" src="https://raw.githubusercontent.com/dannagle/PacketSender/1a18642ac2eb47992a79f1f56cdf19cb9ce296cc/src/languages/english-flag.png">

This release was sponsored by [NagleCode](https://dannagle.com/). If you would like to also support this project, please visit [GitHub Sponsors](https://github.com/sponsors/dannagle).

Thank you to all the individual private donors that continue to make this project possible.

Packet Sender now supports translations to:
- English
- Spanish
- German
- French

More importantly, the code has been reworked so more languages can be easily added. I hope to support more languages in future releases.

Other updates:

- You can now launch the starter panel in read-only mode from the command line using `--starterpanel` option. This will be very helpful for control systems.
- The `{{TIME}}` and `{{DATE}}` format can be customized.
- The Windows version is now running Qt6. This should help with 4K rendering.
- Traffic log can now be sorted
- A starter panel is now bundled that demonstrates sending and panel transitions
- Can now bypass the Linux gui test (using `--gui`)

Thank you, and happy sending and receiving!

<a id="spanish"></a>
<hr>
<img width="200" src="https://raw.githubusercontent.com/dannagle/PacketSender/1a18642ac2eb47992a79f1f56cdf19cb9ce296cc/src/languages/spanish-flag.png">

Esta versión ha sido patrocinada por [NagleCode](https://dannagle.com/). Si desea apoyar también este proyecto, visite [GitHub Sponsors](https://github.com/sponsors/dannagle).

Gracias a todos los donantes particulares que siguen haciendo posible este proyecto.

Packet Sender ahora tiene traducciones en:
- Inglés
- Español
- Alemán
- Francés

Y lo que es más importante, se ha reelaborado el código para que se puedan añadir fácilmente más idiomas. En próximas versiones se espera dar soporte a más idiomas.

Otras actualizaciones:

- Ahora puede lanzar el panel de inicio en modo de sólo lectura desde la línea de comandos utilizando la opción `--starterpanel`. Esto será muy útil para los sistemas de control.
- Los formatos `{{TIME}}` y `{{DATE}}` pueden poersonalizarse.
- La versión para Windows ejecuta ahora Qt6. Esto debería ayudar con el renderizado en 4K.
- Ahora se puede ordenar el registro de tráfico
- Ahora se incluye un panel de inicio que demuestra el envío y las transiciones del panel
- Ahora se puede evitar la prueba gui de Linux (usando `--gui`)

¡Gracias, y Felices Envíos y Recepciones!

<a id="german"></a>
<hr>
<img width="200" src="https://raw.githubusercontent.com/dannagle/PacketSender/1a18642ac2eb47992a79f1f56cdf19cb9ce296cc/src/languages/german-flag.png">

Diese Version wurde von [NagleCode](https://dannagle.com/) gesponsert. Wenn auch Sie dieses Projekt unterstützen möchten, besuchen Sie bitte [GitHub Sponsors](https://github.com/sponsors/dannagle).

Vielen Dank an alle privaten Spender, die dieses Projekt weiterhin möglich machen.

Packet Sender unterstützt jetzt Übersetzungen in:
- Englisch
- Spanisch
- Deutsch
- Französisch

Noch wichtiger ist, dass der Code überarbeitet wurde, so dass weitere Sprachen leicht hinzugefügt werden können. Ich hoffe, dass ich in zukünftigen Versionen mehr Sprachen unterstützen kann.

Andere Aktualisierungen:

- Sie können nun das Starterpanel im Nur-Lese-Modus von der Kommandozeile aus mit der Option `--starterpanel` starten. Dies wird für Kontrollsysteme sehr hilfreich sein.
- Das Format von `{{TIME}}` und `{DATE}}` kann angepasst werden.
- Die Windows-Version läuft jetzt mit Qt6. Dies sollte beim 4K-Rendering helfen.
- Das Verkehrslog kann jetzt sortiert werden
- Ein Starter-Panel ist jetzt gebündelt, das das Senden und die Panel-Übergänge demonstriert
- Der Linux-Gui-Test kann nun umgangen werden (mit `--gui`)

Vielen Dank, und viel Spaß beim Senden und Empfangen!

<a id="french"></a>
<hr>
<img width="200" src="https://raw.githubusercontent.com/dannagle/PacketSender/1a18642ac2eb47992a79f1f56cdf19cb9ce296cc/src/languages/french-flag.png">

Cette version a été sponsorisée par [NagleCode](https://dannagle.com/). Si vous souhaitez également soutenir ce projet, veuillez consulter [GitHub Sponsors](https://github.com/sponsors/dannagle).

Merci à tous les donateurs privés qui continuent à rendre ce projet possible.

Packet Sender supporte désormais les traductions vers :
- Anglais
- Espagnol
- Allemand
- français

Plus important encore, le code a été retravaillé afin que d'autres langues puissent être facilement ajoutées. J'espère pouvoir supporter plus de langues dans les prochaines versions.

Autres mises à jour :

- Vous pouvez maintenant lancer le panneau de démarrage en mode lecture seule depuis la ligne de commande en utilisant l'option `--starterpanel`. Cela sera très utile pour les systèmes de contrôle.
- Les formats `{{TIME}}` et `{{DATE}}` peuvent être personnalisés.
- La version Windows utilise maintenant Qt6. Cela devrait aider avec le rendu 4K.
- Le registre de trafic peut maintenant être trié
- Un panneau de démarrage est maintenant fourni qui démontre l'envoi et les transitions du panneau.
- On peut maintenant contourner le test du gui Linux (en utilisant `--gui`)

Merci, et bon envoi et réception !

<hr /><em>This discussion was created from the release <a href='https://github.com/dannagle/PacketSender/releases/tag/v8.3.5'>Now With Translations!</a>.</em>

**Comment by capedra** [comment]
@dannagle 
Hello, congratulations for the development of this great app!! But whenever I try to open the installer Version v8.3.5 for Windows (https://github.com/dannagle/PacketSender/releases/download/v8.3.5/PacketSender_x64_v8.3.5.exe) I get this error:

![Erro 21_03_2023 01_57_52](https://user-images.githubusercontent.com/412434/226522749-935dd7c3-17f5-49fd-952b-ec07da73890a.png)

**Comment by dannagle** [comment]
@capedra Thanks for the heads up! It is fixed now.

Something got messed up in the pipeline. I am not sure what happened, but I re-ran the build, and it worked. Bizarre. 

Windows main installer is now on v8.3.6.

**Reply by dannagle** [reply]
Portable version built fine. It stays at 8.3.5

**Comment by dieseljwt** [comment]
This is all great, although I was using English, anyway. But must I specify my language AND answer the release notes question every time I start up? I missed it if there is a setting somewhere for defaults.

**Reply by dannagle** [reply]
It should remember your answers unless you hit the x at the top right.

Do you have better luck with the portable version?

**Comment by dieseljwt** [comment]
No, I answer both questions, every time. I’ll try the other version.
